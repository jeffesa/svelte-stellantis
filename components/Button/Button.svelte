<script>
	import { createEventDispatcher } from 'svelte'

	export let to = ''
	export let isBlank = false
	export let label = ''
  export let disabled = false
  export let styles = ''

	const dispatch = createEventDispatcher()
	const withSlot = $$props.$$slots
</script>

<!-- <style>
  .withHover {
    @apply transition duration-300 hover:opacity-80;
  }
</style> -->

{#if to}
	<a
		on:click="{() => dispatch('click')}"
    style="{styles}"
		href="{to}"
		class="withHover {$$props.class}"
		target="{isBlank ? '_blank' : '_self'}"
  >
    {label}
  </a>
{:else}
	<button
    class="outline-none {!disabled && 'withHover'} focus:outline-none {$$props.class}"
    style="{styles}"
    disabled="{disabled}"
		on:click="{(evt) => dispatch('click', evt)}"
	>
		{#if withSlot}
			<slot />
    {:else}
      {label}
    {/if}
	</button>
{/if}
