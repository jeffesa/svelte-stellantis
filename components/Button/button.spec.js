import { render, fireEvent } from '@testing-library/svelte'
import Button from './Button.svelte'

test('set a text to button', () => {
	const { getByText } = render(Button, { label: 'foo' })
	expect(getByText('foo')).toBeInTheDocument()
})

test('button default is sm', () => {
	const { container } = render(Button)
	expect(container).toMatchSnapshot()
})

test('the button can be md', () => {
	const { container } = render(Button, { md: true })
	expect(container).toMatchSnapshot()
})

test('the button can be lg', () => {
	const { container } = render(Button, { lg: true })
	expect(container).toMatchSnapshot()
})

test('button dispatch a custom onClick function', async () => {
	const onClick = jest.fn()
	const { container } = render(Button, { onClick })
	const $btn = container.querySelector('button')
	await fireEvent.click($btn)
	expect(onClick).toBeCalled()
})
