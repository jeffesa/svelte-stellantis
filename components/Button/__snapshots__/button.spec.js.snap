// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button default is sm 1`] = `
<body>
  <div>
    <button
      class="sm"
    >
      
    </button>
  </div>
</body>
`;

exports[`the button can be lg 1`] = `
<body>
  <div>
    <button
      class="lg"
    >
      
    </button>
  </div>
</body>
`;

exports[`the button can be md 1`] = `
<body>
  <div>
    <button
      class="md"
    >
      
    </button>
  </div>
</body>
`;
